core/sql/exp/exp_function.cpp

```diff
@@ -7724,5 +7753,354 @@ Lng32 ExFunctionExtractColumns::unpack (void * base, void * reallocator)
    if (extractColList_.unpack(base)) return -1;
    return unpackClause(base, reallocator);
 }
+
+//helper function, convert a string into IPV4 , if valid, it can support leading and padding space
+static Lng32 string2ipv4(char *srcData, Lng32 slen, unsigned int *inet_addr)
+{
+   Int16 i = 0, j = 0 , p=0, leadingspace=0;
+   char buf[16]; 
+   Int16 dot=0;
+
+   if(slen < MIN_IPV4_STRING_LEN ) 
+     return 0;
+
+   unsigned char *ipv4_bytes= (unsigned char *)inet_addr;
+
+   if(srcData[0] == ' ')
+   { 
+     leadingspace++;
+     for(i=1; i< slen; i++)
+     {
+       if(srcData[i] == ' ') leadingspace++;
+       else break; 
+     }
+   }
```

Comment:
This looks correct. A simpler way to do it, though, might be:

char * next = srcData;
while (*next == ' ')
   {
      leadingspace++;
      next++;
   }